src/main/kotlin/com.bugsnag.android.gradle/BugsnagInstallJniLibsTask.kt

```diff
@@ -60,5 +74,93 @@ open class BugsnagInstallJniLibsTask @Inject constructor(
                 .toSet()
             return project.files(files)
         }
+
+        /**
+         * Registers the appropriate subtype to this [project] with the given [name] and
+         * [configurationAction]
+         */
+        internal fun register(
+            project: Project,
+            name: String,
+            configurationAction: BugsnagInstallJniLibsTask.() -> Unit
+        ): TaskProvider<out BugsnagInstallJniLibsTask> {
+            val gradleVersion = project.gradle.versionNumber()
+            return when {
+              gradleVersion >= GradleVersions.VERSION_6 -> {
+                  when {
+                    gradleVersion >= GradleVersions.VERSION_6_6 -> {
+                        project.tasks.register<BugsnagInstallJniLibsTaskGradle66Plus>(name, configurationAction)
+                    }
+                    else -> {
+                        project.tasks.register<BugsnagInstallJniLibsTaskGradle6Plus>(name, configurationAction)
+                    }
+                  }
+              }
+              gradleVersion >= GradleVersions.VERSION_5_3 -> {
+                  project.tasks.register<BugsnagInstallJniLibsTask53Plus>(name, configurationAction)
+              }
+              else -> {
+                  project.tasks.register<BugsnagInstallJniLibsTaskLegacy>(name, configurationAction)
+              }
+            }
+        }
     }
 }
+
```

Comment:
I think the docs and naming below are mostly self explanatory, but one thing I thought was worth mentioning is that it might be better to use inheritance to de-dupe some of the logic here (and in other task facades). Let me know what you think, I'm sort of undecided since the duplication isn't that much

Reply:
I think the current approach works well and it's quite clear that each task supports a range of Gradle versions. A bit of duplication is fine particularly as there are only a couple of subclasses at the moment.